feat(cdk/visit/lambda_code): add print statement for monitoring

DillonRanwala opened the pull request:
Problem:
Successful lambda canary tests do not have any print statements that show up in the Cloudwatch logs for the TestAPI Lambda. It can be difficult to determine which logs contain successful test runs.

Solution:
By adding this print statement, the Cloudwatch logs should display the time and username for any successful tests.

Testing:
By adding this print statement in dev instances, the lambda functionality was not affected.

Issue:
N/A, No open issue.

DillonRanwala committed:
docs(cdk/visit/lambda_code): add print statement to increase
traceability when monitoring lambda canary tests.

Problem:
Successful lambda canary tests do not have any print statements
that show up in the Cloudwatch logs for the TestAPI Lambda. It can be
difficult to determine which logs contain successful test runs.

Solution:
By adding this print statement, the Cloudwatch logs should display the time and
username for any successful tests.

Testing:
By adding this print statement in dev instances, the lambda
functionality was not affected.

Issue:
N/A, No open issue.

ophilli commented on cdk/visit/lambda_code/test_api/test_api.py:
```diff
@@ -84,14 +84,17 @@ def handle_test_api(self):
             raise Exception("Register API Call Failed")
 
 
+        print("Canary Successful at " + str(dt_string) + " for username: " + str(register_data_dict["username"]))
```
I think the cloud watch logs will include the time stamp so there’s no need to bake it in here. Can you check and amend this PR if the time stamps are already available?

I also want to call out that it is very important to be careful when considering logging customer data. Logging a request id or a hash might be good ways to accomplish the same thing without making your logs as sensitive as your database. Access to the logs is already restricted, but it’s good to be conscientious about what kind of data you put where. 

Can you talk with Prof. Russell about your plan for user checkin system’s data classifications and privacy policy? I’d be happy to hop on a call and provide some general advice but this is ultimately up to y’all to lead.

DillonRanwala replied:
I have adjusted the commit message. Also, Cloudwatch does have the timestamp, so I added a new commit to remove that from the print statement. 

With regards to the username, this log will only take place from the TestAPI lambda which generates its own dummy requests and usernames. The username referenced in that print statement will only be the username of a Canary test item. This should never touch the actual customer data and the only thing in the logs would be the Canary's username (in the form CANARY_TEST_<SOME_DATETIME>) which will be deleted after a few minutes from Dynamo ttl. 

I didn't believe that this information would be sensitive or need to be restricted/hashed, but let me know if you think we still might want to restrict that data.

ophilli replied:
Logging the canary test username will certainly be fine - I did not understand that context.

ophilli reviewed (changes requested):
Adding logging to the lambda will make it way easier to understand the system, this is a good idea.

Can you update your commit message so that the subject line is shorter? I see that it word wrapped. Also, I don’t think this is a “docs” change, seems like a “feat” to me. I also left some comments about the specifics of what you’re logging.

DillonRanwala committed:
feat(cdk/visit/lambda_code): change print statement for log

Problem:
This print statement contained a redundant reference to the time of
execution. Cloudwatch already contains this information in its logs.

Solution:
Change the print statement to remove any reference to the current time
of execution.

Testing:

Dev environments saw no change.

Issue:
N/A, No open issue.

ophilli commented:
@DillonRanwala when you updated the commit message it looks like you added a second commit instead of amending the first one. I'm going to [squash & merge](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/incorporating-changes-from-a-pull-request/about-pull-request-merges#squash-and-merge-your-commits) this PR which will clean that up a little bit, but in the future can you try to amend your commits so that if we rebase and merge the commit log will be exactly how you want it?

I'm not sure how trivial that is to do on GitHub. If it's a huge pain please let me know.

DillonRanwala commented:
> 

I see, I tried editing the message through Github, but I suppose that created a new commit. I will keep this in mind for any future pull requests, thanks for letting me know.
